### ISSUE TITLE
安装PaddlePaddle后进行验证阶段出现错误

### ISSUE THREAD

**Issue by qq345481843** [PaddlePaddle/Paddle#23084]
系统版本：ubuntu 18
cuda版本：10.0
cudnn版本：7.4
显卡：Tesla v100 x3

安装完后，按照教程验证输入
>>> import paddle.fluid
>>> paddle.fluid.install_check.run_check()
报错
Running Verify Paddle Program ... 
/home/jhwang/.local/lib/python3.6/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
  "The following exception is not an EOF exception.")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/install_check.py", line 124, in run_check
    test_simple_exe()
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/install_check.py", line 120, in test_simple_exe
    exe0.run(startup_prog)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 783, in run
    six.reraise(*sys.exc_info())
  File "/home/cyye/.local/lib/python3.6/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 778, in run
    use_program_cache=use_program_cache)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 831, in _run_impl
    use_program_cache=use_program_cache)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 905, in _run_program
    fetch_var_name)
https://github.com/PaddlePaddle/Paddle/issues/new

之后我输入export CUDA_VISIBLE_DEVICES=0指定GPU后，就验证成功了，但是之后我开始训练模型后又出现了上面的错误

希望能得到解决方案，谢谢。

**Comment by qq345481843** [comment]
020-03-18 18:26:41,820-INFO: 60 samples in file dataset/fruit/val.txt
2020-03-18 18:26:41,822-INFO: places would be ommited when DataLoader is not iterable
W0318 18:26:42.999328 36406 device_context.cc:237] Please NOTE: device: 0, CUDA Capability: 70, Driver API Version: 10.0, Runtime API Version: 10.0
W0318 18:26:43.012061 36406 device_context.cc:245] device: 0, cuDNN Version: 7.4.
W0318 18:26:43.012099 36406 device_context.cc:271] WARNING: device: 0. The installed Paddle is compiled with CUDNN 7.6, but CUDNN version in your machine is 7.4, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
2020-03-18 18:26:44,389-INFO: Found /home/cyye/.cache/paddle/weights/yolov3_mobilenet_v1
2020-03-18 18:26:44,389-INFO: Loading parameters from /home/cyye/.cache/paddle/weights/yolov3_mobilenet_v1...
2020-03-18 18:26:44,503-WARNING: /tmp/tmppvzg3ij9/yolov3_mobilenet_v1.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
2020-03-18 18:26:44,503-WARNING: /tmp/tmppvzg3ij9/yolov3_mobilenet_v1.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
2020-03-18 18:26:44,695-INFO: 240 samples in file dataset/fruit/train.txt
2020-03-18 18:26:51,794-INFO: places would be ommited when DataLoader is not iterable
I0318 18:26:52.456447 36406 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 1 cards are used, so 1 programs are executed in parallel.
I0318 18:26:52.506904 36406 build_strategy.cc:365] SeqOnlyAllReduceOps:0, num_trainers:1
I0318 18:26:52.578439 36406 parallel_executor.cc:307] Inplace strategy is enabled, when build_strategy.enable_inplace = True
I0318 18:26:52.620237 36406 parallel_executor.cc:375] Garbage collection strategy is enabled, when FLAGS_eager_delete_tensor_gb = 0
/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
  "The following exception is not an EOF exception.")
Traceback (most recent call last):
  File "tools/train.py", line 323, in <module>
    main()
  File "tools/train.py", line 233, in main
    outs = exe.run(compiled_train_prog, fetch_list=train_values)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 783, in run
    six.reraise(*sys.exc_info())
  File "/home/cyye/.local/lib/python3.6/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 778, in run
    use_program_cache=use_program_cache)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 843, in _run_impl
    return_numpy=return_numpy)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/executor.py", line 677, in _run_parallel
    tensors = exe.run(fetch_var_names)._move_to_list()
paddle.fluid.core_avx.EnforceNotMet: 

--------------------------------------------
C++ Call Stacks (More useful to developers):
--------------------------------------------
0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
2   paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, float>::Compute(paddle::framework::ExecutionContext const&) const
3   std::_Function_handler<void (paddle::framework::ExecutionContext const&), paddle::framework::OpKernelRegistrarFunctor<paddle::platform::CUDAPlace, false, 0ul, paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, float>, paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, double>, paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, paddle::platform::float16> >::operator()(char const*, char const*, int) const::{lambda(paddle::framework::ExecutionContext const&)#1}>::_M_invoke(std::_Any_data const&, paddle::framework::ExecutionContext const&)
4   paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, paddle::platform::Place const&, paddle::framework::RuntimeContext*) const
5   paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, paddle::platform::Place const&) const
6   paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, paddle::platform::Place const&)
7   paddle::framework::details::ComputationOpHandle::RunImpl()
8   paddle::framework::details::FastThreadedSSAGraphExecutor::RunOpSync(paddle::framework::details::OpHandleBase*)
9   paddle::framework::details::FastThreadedSSAGraphExecutor::RunOp(paddle::framework::details::OpHandleBase*, std::shared_ptr<paddle::framework::BlockingQueue<unsigned long> > const&, unsigned long*)
10  std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, void> >::_M_invoke(std::_Any_data const&)
11  std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&)
12  ThreadPool::ThreadPool(unsigned long)::{lambda()#1}::operator()() const

------------------------------------------
Python Call Stacks (More useful to users):
------------------------------------------
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/framework.py", line 2525, in append_op
    attrs=kwargs.get("attrs", None))
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/layer_helper.py", line 43, in append_op
    return self.main_program.current_block().append_op(*args, **kwargs)
  File "/home/cyye/.local/lib/python3.6/site-packages/paddle/fluid/layers/nn.py", line 2663, in batch_norm
    type="batch_norm", inputs=inputs, outputs=outputs, attrs=attrs)
  File "/home/cyye/jhwang/PaddleDetection/ppdet/modeling/backbones/mobilenet.py", line 98, in _conv_norm
    moving_variance_name=bn_name + '_variance')
  File "/home/cyye/jhwang/PaddleDetection/ppdet/modeling/backbones/mobilenet.py", line 159, in __call__
    input, 3, int(32 * scale), 2, 1, name=self.prefix_name + "conv1")
  File "/home/cyye/jhwang/PaddleDetection/ppdet/modeling/architectures/yolov3.py", line 61, in build
    body_feats = self.backbone(im)
  File "/home/cyye/jhwang/PaddleDetection/ppdet/modeling/architectures/yolov3.py", line 156, in train
    return self.build(feed_vars, mode='train')
  File "tools/train.py", line 116, in main
    train_fetches = model.train(feed_vars)
  File "tools/train.py", line 323, in <module>
    main()

----------------------
Error Message Summary:
----------------------
Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
  - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
  - Recommended issue content: all error stack information
  [Hint: CUDNN_STATUS_BAD_PARAM] at (/paddle/paddle/fluid/operators/batch_norm_op.cu:287)
  [operator < batch_norm > error]

这是我指定单个GPU训练后报错信息

**Comment by JepsonWong** [comment]
请问是单卡 run_check 验证成功？但单卡运行程序失败？那您是不是看看代码哪里写错了？

您有没有试过export CUDA_VISIBLE_DEVICES指定多卡验证呢

**Comment by qq345481843** [comment]
可以运行了，应该是当时在验证的时候后台进程没清除，显卡被占用了。谢谢答复。
